frontend/packages/ceph-storage-plugin/src/components/ocs-install/create-form.tsx

```diff
@@ -1,21 +1,86 @@
 import * as React from 'react';
-import { Alert, Title } from '@patternfly/react-core';
-import { K8sResourceKind, K8sKind } from '@console/internal/module/k8s';
+import * as _ from 'lodash';
+import { match } from 'react-router';
+import { Alert, Title, ActionGroup, Button } from '@patternfly/react-core';
+import { NodeKind, k8sPatch, k8sCreate, referenceForModel } from '@console/internal/module/k8s';
 import { ListPage } from '@console/internal/components/factory';
 import { NodeModel } from '@console/internal/models';
-import { ClusterServiceVersionKind } from '@console/operator-lifecycle-manager/src/types';
-import { NodeList } from './node-list';
-
+import { hasLabel, getName } from '@console/shared';
+import {
+  withHandlePromise,
+  HandlePromiseProps,
+  history,
+  FieldLevelHelp,
+  ButtonBar,
+} from '@console/internal/components/utils';
+import { ocsRequestData, labelTooltip, minSelectedNode } from '../../constants/ocs-install';
+import { OCSServiceModel } from '../../models';
+import { OCSStorageClassDropdown } from '../modals/storage-class-dropdown';
+import { OSDSizeDropdown } from '../../utils/osd-size-dropdown';
+import NodeTable from './node-list';
 import './ocs-install.scss';
+import { cephStorageLabel } from '../../selectors';
 
-export const CreateOCSServiceForm: React.FC<CreateOCSServiceFormProps> = ({
-  namespace,
-  clusterServiceVersion,
-}) => {
-  const ListComponent = React.useCallback(
-    (nodeProps) => <NodeList {...nodeProps} ocsProps={{ namespace, clusterServiceVersion }} />,
-    [namespace, clusterServiceVersion],
+const makeLabelNodesRequest = (selectedNodes: NodeKind[]): Promise<NodeKind>[] => {
+  const patch = [
+    {
+      op: 'add',
+      path: '/metadata/labels/cluster.ocs.openshift.io~1openshift-storage',
+      value: '',
+    },
+  ];
+  return _.reduce(
+    selectedNodes,
+    (accumulator, node) => {
+      return hasLabel(node, cephStorageLabel)
+        ? accumulator
+        : [...accumulator, k8sPatch(NodeModel, node, patch)];
+    },
+    [],
   );
+};
+
+const makeOCSRequest = (
+  selectedData: NodeKind[],
+  storageClass: string,
+  osdSize: string,
+): Promise<any> => {
+  const promises = makeLabelNodesRequest(selectedData);
+  const ocsObj = _.cloneDeep(ocsRequestData);
+  ocsObj.spec.storageDeviceSets[0].dataPVCTemplate.spec.storageClassName = storageClass;
```

Comment:
This needs to be moved inside the `CreateOCSServiceForm` component as for baremetal, there is a check for storageclass's provisioner and this check is inside `CreateOCSServiceForm`.

Reply:
Got it. You are basically calling it with all the params and returning it.

Reply:
I am not following you. Perhaps a suggested change might make things clearer for me.

Reply:
Please ignore. I got confused.